src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/spotify/PlayerLogic.java

```diff
@@ -0,0 +1,423 @@
+					.getUsersAvailableDevices().build();
+			try {
+				Device[] devices = getUsersAvailableDevicesRequest.execute();
+				if (deviceNumber < devices.length) {
+					this.deviceID = devices[deviceNumber].getId();
+					return devices[deviceNumber].getName();
+				}
+				return "This device was not found";
+			} catch (SpotifyWebApiException | IOException e) {
+				e.printStackTrace();
+				System.err.println(e);
+				return "A problem has occurred";
+			}
+
+		}
+		return "Please init the spotify API";
+	}
+
+	/**
+	 * this call the searchAnaything method in the Search class
+	 * 
+	 * @param searchText
+	 * @param type
+	 *            artist, track, playlist, album
+	 * @param limit
+	 *            how many results maximal searched for
+	 * @return one output String with all results
+	 */
+	public String search(String searchText, String type, int limit) {
+		if (checkPlayerState() == null) {
+			this.actualSearchResult = this.search.SearchAnything(searchText, type, limit);
+			String resultString = "";
+			for (int i = 0; i < this.actualSearchResult.size(); i++) {
+				resultString = resultString + "\n" + this.actualSearchResult.get(i)[1];
+			}
+			for (String[] result : this.actualSearchResult) {
+				resultString = resultString + "\n" + result[1];
+			}
+			return resultString;
+		} else {
+			return checkPlayerState();
+		}
+	}
+
+	/**
+	 * this play method play a featured playlist from spotify
+	 * @return
+	 */
+	public String play() {
+		ArrayList<String[]> playLists = this.search.getFeaturedPlaylists();
+		if (playLists != null) {
+			if (1 < this.search.getFeaturedPlaylists().size()) {
+				playListFromUri(playLists.get(1)[0]);
+				return playLists.get(1)[1];
+			}
+		}
+		return "no featured playlist found";
+	}
+
+	/**
+	 * this method play the item that searched before. Use only after a search
+	 * 
+	 * @param songNumber
+	 *            number of the item form the search before
+	 * @return
+	 */
+	public String play(int songNumber) {
+		if (this.actualSearchResult != null && songNumber < this.actualSearchResult.size()) {
+			if (this.actualSearchResult.get(songNumber)[2].equals("track")) {
+				playSongFromUri(this.actualSearchResult.get(songNumber)[0]);
+				return this.actualSearchResult.get(songNumber)[1];
+			} else {
+				playListFromUri(this.actualSearchResult.get(songNumber)[0]);
+				return this.actualSearchResult.get(songNumber)[1];
+			}
+		} else {
+			return "Please search before you call this or choose a smaller number";
+		}
+	}
+
+	/**
+	 * resume the actual playback
+	 * 
+	 * @return short String about the execute action
+	 */
+	public String resume() {
+		if (checkPlayerState() == null) {
+			StartResumeUsersPlaybackRequest startResumeUsersPlaybackRequest = this.auth.getSpotifyApi()
+					.startResumeUsersPlayback().device_id(this.deviceID).build();
+			try {
+				startResumeUsersPlaybackRequest.execute();
+				return "Playback resume";
```

Comment:
the string for the output should be generated in the Speech class, because it is then possible to use the logic without the speechcommands